pages/api/v1/players/index.ts

```diff
@@ -0,0 +1,224 @@
+    AND SeasonID=${season.SeasonID};
+`);
+
+const getGoalieStats = async (league, season) => await query(SQL`
+  SELECT *
+  FROM player_goalie_stats_rs
+  WHERE LeagueID=${+league}
+    AND SeasonID=${season.SeasonID};
+`);
+
+const getPlayerInfo = (player: MasterPlayer) => ({
+  id: player.PlayerID,
+  league: player.LeagueID,
+  season: player.SeasonID,
+  name: player['Last Name'],
+  team: player.TeamID,
+  position: player.position
+});
+
+export default async (
+  req: NextApiRequest,
+  res: NextApiResponse
+): Promise<void> => {
+  await use(req, res, cors);
+
+  const { league = 0, season: seasonid } = req.query;
+
+  const [season] =
+    (!Number.isNaN(+seasonid) && [{ SeasonID: +seasonid }]) ||
+    (await query(SQL`
+      SELECT DISTINCT SeasonID
+      FROM player_master
+      WHERE LeagueID=${+league}
+      ORDER BY SeasonID DESC
+      LIMIT 1
+  `));
+
+  const basePlayerData = await getBasePlayerData(league, season);
+  const skaterStats = await getSkaterStats(league, season);
+  const goalieStats = await getGoalieStats(league, season);
+  const combinedPlayerData = basePlayerData.map(player => {
+    const position = ['G', 'LD', 'RD', 'LW', 'C', 'RW'][
+      [
+        player.G,
+        player.LD,
+        player.RD,
+        player.LW,
+        player.C,
+        player.RW,
+      ].indexOf(20)
+    ];
+
+    if (position === 'G') {
+      return {
+        baseData: player,
+        position,
+        stats: goalieStats.find(goalie => goalie.PlayerID === player.PlayerID)
+      };
+    }
+    return {
+      baseData: player,
+      position,
+      stats: skaterStats.find(skater => skater.PlayerID === player.PlayerID)
+    };
+  });
+
+  const parsed = combinedPlayerData.map(player => {
+    const playerInfo = getPlayerInfo(player.baseData);
+
+    if (player.position === 'G') {
+      return {
+        ...playerInfo,
+        stats: {
+          gamesPlayed: player.stats.GP,
+          minutes: player.stats.Minutes,
+          wins: player.stats.Wins,
+          losses: player.stats.Losses,
+          ot: player.stats.OT,
+          shotsAgainst: player.stats.ShotsAgainst,
+          saves: player.stats.Saves,
+          goalsAgainst: player.stats.GoalsAgainst,
+          gaa: player.stats.GAA,
+          shutouts: player.stats.Shutouts,
+          savePct: player.stats.SavePct,
+          gameRating: player.stats.GameRating
+        },
+        ratings: {
+          blocker: player.Blocker,
+          glove: player.Glove,
+          passing: player.GPassing,
+          pokeCheck: player.GPokecheck,
+          positioning: player.GPositioning,
+          rebound: player.Rebound,
+          recovery: player.Recovery,
+          puckhandling: player.GPuckhandling,
+          lowShots: player.LowShots,
+          reflexes: player.Reflexes,
+          skating: player.GSkating,
+          mentalToughness: player.MentalToughness,
+          goalieStamina: player.GoalieStamina
+        }
+      };
+    }
+
+    return {
```

Comment:
This is a massive amount of data. Can we add a "full" parameter which lets us only return important data rather than all advanced stats and ratings. I dont see us needing at least 70-80% of this data for the most part.

Reply:
This big blob of data contains all we'd need to make a page that contains all stats, adv.stats and the ratings. Figured we could use the same kind of selector we have on the Standings page to quickly switch between the three. But that is if you agree that it's a decent approach. Otherwise we can easily split this up and only return one type of data per request. I'm not able to test just how heavy this would be, so it might be a terrible idea. WDYT?

Reply:
Well the idea would be to have a condensed version that would be default but an option for a full view if necessary. The stats page is more of a leaderboard page if this is what is being made for that

Reply:
I see. We could do a similar thing as for the schedule. Allow us to send in a "type" parameter. It could either be "stats", "advanced", "ratings", or "full". And then have this return whichever type of data has been requested. I'll work on adding this logic early next week.

Reply:
Updated! @esilverm let me know what you think.